### PR TITLE
Disable deprecation warnings in parts of the library that use deprecated headers

### PR DESCRIPTION
**Pull request by traversaro**
Some parts of the library are deprecated themselves, so it make sense that they use deprecated headers.
Furthermore, bindings wrap also methods until they are removed, so it is ok if bindings wrap deprecated methods.

I was trying to fix completely https://github.com/robotology/idyntree/issues/312 , but I was unable to find a way to suppress warnings induced by the `#warning` preprocessor directive. @diegoferigo any idea on how to fix it?

### CODE REVIEW

**Comment by diegoferigo** [comment]
Did you try adding `-Wno-deprecated` in addition to `-Wno-deprecated-declarations`? Some links in [the comment](https://github.com/robotology/idyntree/pull/310#discussion_r120109122) @francesco-romano wrote.

**Comment by traversaro** [comment]
Thanks @diegoferigo , the latest commit should be able to fix https://github.com/robotology/idyntree/issues/312 .

File: cmake/iDynTreeOptions.cmake [hunk 1]
```diff
@@ -131,3 +131,18 @@ macro(idyntree_enable_cxx11)
...
+macro(idyntree_disable_deprecation_warnings)
+  if(MSVC)
+    add_definitions(/wd4996)
+  else()
+    check_cxx_compiler_flag("-Wno-deprecated-declarations" CXX_HAS_WNO_DEPRECATED_DECLARATIONS)
+    if(CXX_HAS_WNO_DEPRECATED_DECLARATIONS)
+      add_definitions(-Wno-deprecated-declarations)
+    endif()
+    check_cxx_compiler_flag("-Wno-deprecated" CXX_HAS_WNO_DEPRECATED)
+    if(CXX_HAS_WNO_DEPRECATED_DECLARATIONS)
```
Comment (francesco-romano) [inline cmake/iDynTreeOptions.cmake]: This `if` is wrong. It should be `if(CXX_HAS_WNO_DEPRECATED)`

Comment (traversaro) [reply cmake/iDynTreeOptions.cmake]: Good catch, fixed now.

Comment (traversaro) [reply cmake/iDynTreeOptions.cmake]: Fixed in https://github.com/robotology/idyntree/pull/358/commits/cef7633fe96242ed3cc4160a96c9c6fcda3b24b5 .

**Pull request merged**